NuRadioReco/utilities/trace_utilities.py

```diff
@@ -432,4 +435,206 @@ def split_trace_noise_rms(trace, segments=4, lowest=2):
+
+        # Find the maximum value of the smoothed waveform
+        max_bin = np.argmax(channel_wf)
+        max_val = channel_wf[max_bin]
+
+        root_power_ratio = max_val / noise_rms
+
+    return root_power_ratio
+
+def get_hilbert_envelope(trace):
+    """
+    Calculates the Hilbert envelope of given waveform.
+
+
+    Parameters
+    ----------
+    trace : waveform trace of given channel
+        the 1d array array containing trace of a channel
+    times:
+        time array of a channel
+    Returns
+    -------
+    float
+        Hilbert envelope of the waveform.
+    """
+
+    # Get the Hilbert envelope of the waveform trace
+    envelope = np.abs(hilbert(trace))
+
+    return envelope
+
+def get_impulsivity(trace):
+    """
+    Calculate the impulsivity of a signal.
+
+    This function computes the impulsivity of a signal by performing a Hilbert transform
+    to obtain the analytic signal, calculating the envelope, and then determining the
+    cumulative distribution function (CDF) of the sorted envelope values based on their
+    closeness to the maximum value. The average of the CDF is then scaled and returned
+    as the impulsivity measure.
+
+    Parameters
+    ----------
+    trace : waveform of given channel
+        the 1d array array containing trace of a channel
+
+    Returns
+    -------
+    float
+        The impulsivity measure of the signal, scaled between 0 and 1.
+    """
+
+    envelope = get_hilbert_envelope(trace)
+    maxv = np.argmax(envelope)
+    power_indexes = np.arange(len(envelope)) ## just a list of indices the same length as the array
+    closeness = list(
+        np.abs(power_indexes - maxv)
+    )  ## create an array containing index distance to max voltage (lower the value, the closer it is)
+
+    sorted_power = [x for _, x in sorted(zip(closeness, envelope))]
+    cdf = np.cumsum(sorted_power)
+    cdf = cdf / cdf[-1]
+
+    cdf_avg = (np.mean(np.asarray([cdf])) * 2.0) - 1.0
+
+    if cdf_avg < 0:
+        cdf_avg = 0.0
+    return cdf_avg
+
+def get_entropy(trace, n_hist_bins = 50):
+    """
+    Calculate the shannon entropy of a signal.
+
+
+    Parameters
+    ----------
+    trace : waveform of given channel
+        the 1d array array containing trace of a channel
+    Returns
+    -------
+    float
+        The entropy (randomness) measure of the signal.
+    """
+
+    # Step 1: Discretize the signal into bins
+    # If density = True, the result is the value of the probability density function at the bin,
+    # normalized such that the integral over the range is 1.
+    hist, bin_edges = np.histogram(trace, bins = n_hist_bins, density = True)
+
+    # Step 2: Calculate the probability distribution (normalized)
+    probabilities = hist / np.sum(hist)
+
+    # Step 3: Calculate Shannon Entropy
+    # Using base = 2 for entropy in bits
+    signal_entropy = entropy(probabilities, base = 2)
+
+    return signal_entropy
+
+def get_kurtosis(trace):
```

Comment:
@fschlueter,  to answer your question about what makes the kurtosis so low for the noise waveform, I printed all the variables used in the kurtosis calculation. I think the fourth moment is responsible for the differences. 
![image](https://github.com/user-attachments/assets/9c855dc6-edad-4242-ad9b-5b0026031178)